packages/state-manager/src/sagas/users/userProfile/saveUserProfile.saga.ts

```diff
@@ -0,0 +1,68 @@
+import { type PayloadAction } from '@reduxjs/toolkit'
+import { call, select, apply, put } from 'typed-redux-saga'
+import { arrayBufferToString } from 'pvutils'
+import * as Block from 'multiformats/block'
+import * as dagCbor from '@ipld/dag-cbor'
+import { sha256 } from 'multiformats/hashes/sha2'
+
+import { signData, loadPrivateKey, pubKeyFromCsr } from '@quiet/identity'
+import { UserProfile, UserProfileData, SocketActionTypes } from '@quiet/types'
+
+import { config } from '../../users/const/certFieldTypes'
+import { identitySelectors } from '../../identity/identity.selectors'
+import logger from '../../../utils/logger'
+import { usersActions } from '../users.slice'
+import { type Socket, applyEmitParams } from '../../../types'
+
+const log = logger('saveUserProfileSaga')
+
+const fileToBase64String = (file: File): Promise<string> => {
+  return new Promise<string>((resolve, reject) => {
+    const reader = new FileReader()
+
+    reader.onload = () => {
+      if (reader.result) {
+        resolve(reader.result as string)
+      } else {
+        reject(new Error(`Failed to read file, result: ${reader.result}`))
+      }
+    }
+    reader.onerror = reject
+    reader.readAsDataURL(file)
+  })
+}
+
+export function* saveUserProfileSaga(socket: Socket, action: PayloadAction<{ photo?: File }>): Generator {
+  const identity = yield* select(identitySelectors.currentIdentity)
+
+  if (!identity?.userCsr || !action.payload.photo) {
```

Comment:
I'm not sure if this is the case so don't bother if it's not important in this context but if you want to make sure user is a part of a community you should not only check the presence of csr, but also a replication of the community's metadata. On current develop there's a "communityMembership" selector in the identity structure